frontends/web/src/routes/bitsurance/dashboard.tsx

```diff
@@ -62,40 +67,57 @@ const AccountStatusIcon = ({ status }: TAccountStatusIconProps) => {
 export const BitsuranceDashboard = ({ accounts }: TProps) => {
   const { t } = useTranslation();
   const mounted = useMountedRef();
-  const [insuredAccounts, setInsuredAccounts] = useState<TAccountDetails[]>([]);
   const [balances, setBalances] = useState<Balances>();
-  const [activeAccountCodes, setActiveAccountCodes] = useState<AccountCode[]>([]);
+  const [insurances, setInsurances] = useState<TInsurancesByCode>();
+  const [accountsByKeystore, setAccountsByKeystore] = useState<TAccountsByKeystore[]>();
+
+  // anyAccountInsured returns true if any of the accounts belonging
+  // to a certain keystore is contained in the insurances map.
+  const anyAccountInsured = (keystore: TAccountsByKeystore) => {
+    if (keystore.accounts?.length && insurances) {
+      return keystore.accounts.map(account => account.code)
+        .some(accountCode => insurances[accountCode]);
+    }
```

Comment:
just curious, isnt' there a insured key in accounts?

why not do smth like

```js
return accounts.some(account => account.bitsuranceStatus === 'insured')
```

Reply:
Good point! Actually we can't just check for 'insured' as in the dashboard we display all the possible insurance status. But I simplified the function just checking if account.bitsuranceStatus is not empty! See https://github.com/digitalbitbox/bitbox-wallet-app/compare/c9b2e51361be6a0921614544a761eddbf656c29a..798c4ff0f94a981c1aab27c6c65327394e7ff92f